src/components/RouletteMain/RouletteMain.jsx

```diff
@@ -31,7 +33,12 @@ function RouletteMain(props) {
     const [isMovieDetailsVisible, setMovieDetailsVisible] = useState(false);
     const [editMovieSelection, setEditMovieSelection] = useState({});
     const [deleteMovieSelection, setDeleteMovieSelection] = useState({});
-    const [movieDetailSelection, setMovieDetailSelection] = useState({});
+ 
+    var isSearch = false;
```

Comment:
You should remember, that variables in functional components are not saving between invocations. That's why you need to use **useState** hook. Your solution does not work for me: when I open page by **http://localhost:8080/search/sfsdfsdsds** URL I have infinite loop of requests. To solve this problem in your architecture you need to use **isSearch** as state variable:
```
    const [isSearch, setIsSearch] = useState(false);
```